File: LeetCode/03-longest-substring-without-repeating-chars/solution.py
def lengthOfLongestSubstring(s: str) -> int:
    if s == "":
        return 0
    charIndex: dict[str, int] = {}
    maxlength: int = 0
    left: int = 0

    for right in range(len(s)):
        if s[right] in charIndex and charIndex[s[right]] >= left:
            left = charIndex[s[right]] + 1
        charIndex[s[right]] = right
        maxlength = max(maxlength, right -left +1)

    return maxlength

File: LeetCode/03-longest-substring-without-repeating-chars/solution.ts
function lengthOfLongestSubstring(s: string): number {
  if (s.length == 0) return 0;
  let charIndex: Map<string, number> = new Map<string, number>();
  let maxLength: number = 0;
  let left: number = 0;
  for (const [i, char] of [...s].entries()) {
    if (charIndex.has(char)) {
      left = Math.max(left, charIndex.get(char)! + 1);
    }
    charIndex.set(char, i);
    maxLength = Math.max(maxLength, i - left + 1);
  }
  return maxLength;
}
